packages/@aws-cdk/aws-apigateway/lib/vpc-link.ts

```diff
@@ -37,14 +37,16 @@ export class VpcLink extends Resource {
   public readonly vpcLinkId: string;
 
   constructor(scope: Construct, id: string, props: VpcLinkProps) {
-    super(scope, id);
+    super(scope, id, {
+      physicalName: props.vpcLinkName,
+    });
 
     const cfnResource = new CfnVpcLink(this, 'Resource', {
-      name: props.name || this.node.uniqueId,
+      name: this.physicalName.value || this.node.uniqueId,
```

Comment:
same here. If the default behavior is to allocate a physical name, we need `this.physicalName` to return it.

Reply:
Done.